image-builder-mk3 can nil-deref

### Bug description

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa16af4]

goroutine 73330 [running]:
github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.findTaskLogURL.func1({0xcab5b8, 0xc000213d10}, 0x0, {0xc99980, 0xc0001d8420})
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:375 +0x94
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc0001eae00, 0xc000a166a0)
	github.com/hashicorp/go-retryablehttp@v0.7.0/client.go:597 +0x60d
github.com/hashicorp/go-retryablehttp.(*RoundTripper).RoundTrip(0xc0002036b0, 0xc98420)
	github.com/hashicorp/go-retryablehttp@v0.7.0/roundtripper.go:44 +0x7d
net/http.send(0xc00093ae00, {0xc98420, 0xc0002036b0}, {0xb8f3e0, 0x1, 0x0})
	net/http/client.go:252 +0x5d8
net/http.(*Client).send(0xc000213e00, 0xc00093ae00, {0xc00093ae00, 0x2f, 0x0})
	net/http/client.go:176 +0x9b
net/http.(*Client).do(0xc000213e00, 0xc00093ae00)
	net/http/client.go:725 +0x908
net/http.(*Client).Do(...)
	net/http/client.go:593
github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.findTaskLogURL({0xcab5b8, 0xc000213d10}, {0xc00072b180, 0x3e}, {0xc00028d480, 0x20})
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:382 +0x3bd
github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.listenToHeadlessLogs.func2({0xcab5b8, 0xc000213d10})
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:288 +0x5b
github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.retry({0xcab510, 0xc000f77800}, 0xc000bfbd10, 0xc000bfbc80, 0x3b9aca00, 0xa)
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:172 +0x167
github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.listenToHeadlessLogs({0xcab510, 0xc000f77800}, {0xc00072b180, 0x3e}, {0xc00028d480, 0x20}, 0xc000290fe0)
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:287 +0x209
created by github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.(*buildMonitor).handleStatusUpdate
	github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:138 +0x476
```

### Steps to reproduce

Run with enough load

### Workspace affected

_No response_

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

--- a/components/image-builder-mk3/pkg/orchestrator/monitor.go
+++ b/components/image-builder-mk3/pkg/orchestrator/monitor.go
@@ -372,7 +372,7 @@ func findTaskLogURL(ctx context.Context, ideURL, authToken string) (taskLogURL s
 			// gRPC-web race in supervisor?
 			return true, nil
 		}
-		if resp.StatusCode == http.StatusNotFound {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
 			// We're too quick to connect - ws-proxy doesn't keep up
 			return true, nil
 		}

